pkg/probe/http/http_test.go

```diff
@@ -314,3 +314,39 @@ func TestHTTPProbeChecker_NonLocalRedirects(t *testing.T) {
 		})
 	}
 }
+
+func TestHTTPProbeChecker_HostHeaderPreservedAfterRedirect(t *testing.T) {
+	hostHeader := "www.example.com"
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/redirect":
+			http.Redirect(w, r, "/success", http.StatusFound)
+		case "/success":
+			if r.Host != hostHeader {
```

Comment:
it seems this check will always pass?
`r.Host` probably will be `127.0.0.1...`

Reply:
oops, I fixed the test and added another case to test probe failures.

note that r.URL.Host is different from r.Host, the former is the host part of the url (which in this case will be 127.0.0.1), the latter is the host header value.